cupyx/jit/_cuda_types.py

```diff
@@ -253,6 +275,29 @@ def __str__(self) -> str:
         return f'{self.child_type}*'
 
 
+class ContiguousArray(PointerBase):
+    # used to define ndarray of local memory, in <dtype>[3][4] format
+    def __init__(self, child_type: TypeBase, size: Tuple[int, ...]) -> None:
```

Comment:
Duplicate child dimension lengths are stored in `child_type`. Could you fix `__init__` to only take the outermost length?
```suggestion
    def __init__(self, child_type: TypeBase, size: int) -> None:
```